### PR TITLE
Add support for backup PVC configuration

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
Thank you for contributing to Velero!

# Please add a summary of your change

Implementation for Design https://github.com/vmware-tanzu/velero/pull/7982

# Does your change fix a particular issue?

Fixes https://github.com/vmware-tanzu/velero/issues/7747 and https://github.com/vmware-tanzu/velero/issues/7700

Functional testing pending

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/controller/data_upload_controller.go [hunk 1]
```diff
@@ -828,6 +830,16 @@ func (r *DataUploadReconciler) setupExposeParam(du *velerov2alpha1api.DataUpload
...
 		}
 
+		// check if there is a mapping for source pvc storage class in backupPVC config
+		// if the mapping exists then use the values(storage class, readOnly accessMode)
+		// for backupPVC (intermediate PVC in snapshot data movement) object creation
+		backupPVCStorageClass := du.Spec.CSISnapshot.StorageClass
+		readOnly := false
+		if value, exists := r.backupPVCConfig[du.Spec.CSISnapshot.StorageClass]; exists {
+			backupPVCStorageClass = value.StorageClass
+			readOnly = value.ReadOnly
```
Comment (Lyndon-Li) [inline pkg/controller/data_upload_controller.go:840]: `readOnly = value.ReadOnly` should be outside of the if branch which checks `r.backupPVCConfig[du.Spec.CSISnapshot.StorageClass]`

File: pkg/controller/data_upload_controller.go [hunk 2]
```diff
@@ -828,6 +830,16 @@ func (r *DataUploadReconciler) setupExposeParam(du *velerov2alpha1api.DataUpload
 			return nil, errors.Wrapf(err, "failed to get PVC %s/%s", du.Spec.SourceNamespace, du.Spec.SourcePVC)
 		}
 
+		// check if there is a mapping for source pvc storage class in backupPVC config
```
[the remark this answers is not in the document]
Comment (shubham-pampattiwar) [reply pkg/controller/data_upload_controller.go]: Done, updated the PR.

**Review by Lyndon-Li — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Comment by Lyndon-Li** [comment]
@shubham-pampattiwar 
I just realized one problem after this PR is merged:
```
	backupPVCStorageClass := csiExposeParam.StorageClass
	backupPVCReadOnly := false

	if value, exists := csiExposeParam.BackupPVCConfig[csiExposeParam.StorageClass]; exists {
		backupPVCStorageClass = value.StorageClass
		backupPVCReadOnly = value.ReadOnly
	}
```
Should be modified to:
```
	backupPVCStorageClass := csiExposeParam.StorageClass
	backupPVCReadOnly := false

	if value, exists := csiExposeParam.BackupPVCConfig[csiExposeParam.StorageClass]; exists {
		if value.StorageClass != "" {
			backupPVCStorageClass = value.StorageClass
		}

		backupPVCReadOnly = value.ReadOnly
	}
```

`value.StorageClass` may be empty from the configuration (it is not a must-have field), in this case, we should use the value from `csiExposeParam.StorageClass`, otherwise, the provision will fail.

I add this change to PR #8115. Just for your information.

**Comment by shubham-pampattiwar** [comment]
Thank you @Lyndon-Li !

**Pull request merged**